Feature/footer

CourtneyGosselin opened the pull request:
## Ticket(s)
[DIG-1350](https://candig.atlassian.net/browse/DIG-1350)

## Related PR
[CanDIGv2 PR](https://github.com/CanDIG/CanDIGv2/pull/402)

## Description
Addition of footer to all pages that is responsive to sizing. The footer should include:
- CanDIG website - [CanDIG](https://www.distributedgenomics.ca/) 
- CanDIG GitHub - [CanDIG](https://github.com/CanDIG) 
- CanDIG Platform version number - auto populate from whichever version is deployed
- MoHCCN project link & logo - [Home](https://www.marathonofhopecancercentres.ca/)  
- Funding statement: "CanDIG Receives Funds From:"
     - [TFRI](https://www.tfri.ca/)
     - [UHN DATA](https://uhndata.io/)
     - [BCGSC](https://www.bcgsc.ca/)
     - [C3G](https://computationalgenomics.ca/)

## Screenshots (if appropriate)
### After PR

![screencapture-candig-docker-internal-5080-summary-2024-01-30-18_08_54](https://github.com/CanDIG/candig-data-portal/assets/37649170/a36dd823-4724-460a-9b55-df3901f44706)

## Types of Change(s)

-   [ ] 🪲 Bug fix (non-breaking change that fixes an issue)
-   [x] ✨ New feature (non-breaking change that adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Has it been tested for:

-   [x] My change requires a change to the documentation
-   [x] I have updated the documentation accordingly
-   [x] Prettier linter doesn't return errors
-   [ ] Production branch PR browser testing: Chrome, Firefox, Edge, etc.
-   [x] Locally tested
-   [ ] Dev server tested
-   [ ] Production tested when merging into stable/production branch
-   [x] [Runbook tasks](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) pass locally/on UHN-Dev
-   [x] If visuals have changed, [Runbook](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) has been updated with new screenshots


[DIG-1350]: https://candig.atlassian.net/browse/DIG-1350?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

CourtneyGosselin committed:
Debug push

Signed-off-by: Courtney Gosselin <courtney@gosselin.io>

OrdiNeu committed:
[wip] Fixes from the talk

CourtneyGosselin committed:
Intial layout, image bug

Signed-off-by: Courtney Gosselin <courtney@gosselin.io>

OrdiNeu committed:
[wip] Getting the logo working, but it's not hte irght size

CourtneyGosselin committed:
Final styling and screen sizing

Signed-off-by: Courtney Gosselin <courtney@gosselin.io>

CourtneyGosselin committed:
Email copied work and linter warnings

Signed-off-by: Courtney Gosselin <courtney@gosselin.io>

CourtneyGosselin committed:
Add back theme

Signed-off-by: Courtney Gosselin <courtney@gosselin.io>

CourtneyGosselin committed:
ESLint warnings

Signed-off-by: Courtney Gosselin <courtney@gosselin.io>

CourtneyGosselin committed:
Add version via .env

Signed-off-by: Courtney Gosselin <courtney@gosselin.io>

mshadbolt commented on src/layout/MainLayout/Footer/index.js:
```diff
@@ -0,0 +1,286 @@
+            </ResponsiveSocialsContainer>
+            <ResponsiveFundingContainer style={{ marginRight: '1em' }}>
+                <div style={{ marginRight: '1em' }}>
+                    <p style={{ ...linkText, fontWeight: 'bold', marginBottom: '0.5em' }}>CanDIG Receives Funds from:</p>
```
```suggestion
                    <p style={{ ...linkText, fontWeight: 'bold', marginBottom: '0.5em' }}>CanDIG receives funds from:</p>
```

CourtneyGosselin replied:
Changes were made and responsiveness of graphs moved to their own ticket https://candig.atlassian.net/jira/software/c/projects/DIG/boards/6/backlog?selectedIssue=DIG-1470

mshadbolt reviewed (changes requested):
pulled the branch and rebuilt stack. Looked at in Brave and Chrome - both chromium based browsers

A couple of minor bits of feedback:
* When the summary page loads on my screen, the footer is out of sight of the main page as there is a large gap under the graphs. 
![Screenshot 2024-01-31 at 3 17 46 PM](https://github.com/CanDIG/candig-data-portal/assets/13007987/defb0397-b19f-4691-ad09-fbbb00516c77)
* Maybe we could reduce this spacing so that the footer shows at the bottom of the page. So minimum spacing would be about the same as the gap at the top with the header but when there is enough space it is anchored to the bottom of the screen.
Like:
![Screenshot 2024-01-31 at 3 17 46 PM-edits](https://github.com/CanDIG/candig-data-portal/assets/13007987/33c14219-e1b6-4ad2-8720-508b25ffd14a)

* Can we change the capitalisation of the funding statement to: "CanDIG receives funds from:" (sorry I pointed this out  earlier on the shared screenshots)
* Screenshots above don't show it but I played with the CANDIG_VERSION to see that it displays and it worked for me, not sure if it could be confusing about whether this is the version of the entire stack or the data portal itself.
* Would be nice for the Candig version statement link to the relevant version of the stack? should be easy to add the given version to the url `https://github.com/CanDIG/CanDIGv2/releases/tag/v2.1.1` -> `https://github.com/CanDIG/CanDIGv2/releases/tag/${CANDIG_VERSION}` (however this is done in js that I have no clue about ) this means we do have to ensure the version is input into the .env file correctly when making stable releases to make sure the link isn't broken

CourtneyGosselin committed:
PR review fixes

Changed capilization and linking of versioning. Responsiveness bug moved to new JIRA ticket

Signed-off-by: Courtney Gosselin <courtney@gosselin.io>

mshadbolt reviewed (approved):
Thanks for making the updates! Happy to approve this from a visuals point of view (know nothing about the code so will leave that up to @OrdiNeu )

OrdiNeu commented on src/layout/MainLayout/Footer/index.js:
```diff
@@ -0,0 +1,295 @@
+    fontWeight: 'bold'
+};
+
+const SocialsContainer = styled(Box)(({ theme }) => ({
```
ESLint warning
```suggestion
const SocialsContainer = styled(Box)(({ _ }) => ({
```

OrdiNeu commented on src/layout/MainLayout/Footer/index.js:
```diff
@@ -0,0 +1,295 @@
+    }
+}));
+
+const LinkFrame = styled(Box)(({ theme }) => ({
```
ESLint warning
```suggestion
const LinkFrame = styled(Box)(({ _ }) => ({
```

OrdiNeu commented on src/layout/MainLayout/Footer/index.js:
```diff
@@ -0,0 +1,295 @@
+    color: theme.palette.primary.main
+}));
+
+const ResponsiveFooterContainer = styled('footer')(({ theme }) => ({
```
Could be simplified with the classNames thing that the rest of the components use, but probably not neccecssary

OrdiNeu commented on src/layout/MainLayout/Footer/index.js:
```diff
@@ -0,0 +1,295 @@
+
+        try {
+            document.execCommand('copy');
+            console.log('Email copied to clipboard');
```
```suggestion
```

OrdiNeu commented on src/layout/MainLayout/Footer/index.js:
```diff
@@ -0,0 +1,295 @@
+                navigator.clipboard
+                    .writeText(email)
+                    .then(() => {
+                        console.log('Email copied to clipboard');
```
```suggestion
```
NB: This is usually a mailto: redirect (I believe usually for accessibility?), but to be honest I prefer this copying

CourtneyGosselin replied:
mailto bug not working on all browsers fix for the meantime till that can be fixed for accessibility. This atleast covers dexterity issues when copying but may not be ideal for screen readers. I will add a ticket to JIRA

OrdiNeu reviewed (approved):
Looks good to me

CourtneyGosselin committed:
PR review fixes from Francis

Signed-off-by: Courtney Gosselin <courtney@gosselin.io>

CourtneyGosselin committed:
sidebar fix

Signed-off-by: Courtney Gosselin <courtney@gosselin.io>
